py/client/tests/test_plugin_client.py

```diff
@@ -0,0 +1,37 @@
+#
+#     Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+#
+import unittest
+
+from tests.testbase import BaseTestCase
+
+
+class PluginClientTestCase(BaseTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        server_script = '''
+from deephaven.plot.figure import Figure
+from deephaven import empty_table
+
+source = empty_table(20).update(["Letter = (i % 2 == 0) ? `A` : `B`", "X = 0.1 * i", "Y = randomDouble(0.0, 5.0)"])
+
+plot = Figure().plot_xy(series_name="Random numbers", t=source, x="X", y="Y").show()
+
+source2 = empty_table(20).update(["Letter = (i % 2 == 0) ? `A` : `B`", "X = 0.1 * i", "Y = randomDouble(0.0, 5.0)"])
+
+plot2 = Figure().plot_xy(series_name="Random numbers", t=source2, x="X", y="Y", by=["Letter"]).show()
+    '''
+        self.session.run_script(server_script)
+
+    def test_create(self):
+        plugin_client = self.session.plugin_client(self.session.exportable_objects["plot"])
+        self.assertIsNotNone(plugin_client)
+        payload, refs = next(plugin_client.resp_stream)
+        self.assertGreater(len(payload), 0)
+        self.assertGreater(len(refs), 0)
+        ref = refs[0]
+        self.assertEqual(ref.type, "Table")
```

Comment:
This is probably correct, but I don't have enough context to carefully review this block without pydocs -- or reviewing all of the java.

Reply:
Briefly, this creates two figures, then fetches one, and asserts that it has at least some data (the figure schema itself) and at least one nested object (the table that defines the data for that figure). Not much really new about this, mostly demonstrating that the simple case of fetchable (but not bidi) widgets being read by the python client.

Reply:
I guess as much.  Seems fine.